website/content/v1.2/talos-guides/install/virtualized-platforms/vagrant-libvirt.md

```diff
@@ -0,0 +1,269 @@
+
+Check the status of vagrant VMs:
+
+```bash
+vagrant status
+```
+
+You should see the VMs in "not created" state:
+```
+Current machine states:
+
+control-plane-node-1      not created (libvirt)
+control-plane-node-2      not created (libvirt)
+control-plane-node-3      not created (libvirt)
+worker-node-1             not created (libvirt)
+```
+
+Bring up the vagrant environment:
+```bash
+vagrant up --provider=libvirt
+```
+
+Check the status again:
+```bash
+vagrant status
+```
+
+Now you should see the VMs in "running" state:
+```
+Current machine states:
+
+control-plane-node-1      running (libvirt)
+control-plane-node-2      running (libvirt)
+control-plane-node-3      running (libvirt)
+worker-node-1             running (libvirt)
+```
+
+Find out the IP addresses assigned by the libvirt DHCP by running:
+```bash
+virsh list | grep vagrant | awk '{print $2}' | xargs -t -L1 virsh domifaddr
+```
+
+Output will look like the following:
+```
+virsh domifaddr vagrant_control-plane-node-2
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet0      52:54:00:f9:10:e5    ipv4         192.168.121.119/24
+
+virsh domifaddr vagrant_control-plane-node-1
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet1      52:54:00:0f:ae:59    ipv4         192.168.121.203/24
+
+virsh domifaddr vagrant_worker-node-1
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet2      52:54:00:6f:28:95    ipv4         192.168.121.69/24
+
+virsh domifaddr vagrant_control-plane-node-3
+ Name       MAC address          Protocol     Address
+-------------------------------------------------------------------------------
+ vnet3      52:54:00:03:45:10    ipv4         192.168.121.125/24
+```
+
+Our control plane nodes have the IPs: `192.168.121.203`, `192.168.121.119`, `192.168.121.125` and the worker node has the IP `192.168.121.69`.
+
+Now you should be able to interact with Talos nodes that are in maintenance mode:
+```bash
+talosctl -n 192.168.121.203 disks --insecure
+```
+
+Sample output:
+```
+DEV        MODEL   SERIAL   TYPE   UUID   WWID   MODALIAS                    NAME   SIZE     BUS_PATH
+/dev/vda   -       -        HDD    -      -      virtio:d00000002v00001AF4   -      8.6 GB   /pci0000:00/0000:00:03.0/virtio0/
+```
+
+## Installing Talos
+
+Pick an endpoint IP in the `vagrant-libvirt` subnet but not used by any nodes, for example `192.168.121.100`.
+
+Generate a machine configuration:
+```bash
+talosctl gen config my-cluster https://192.168.121.100:6443
+```
+
+Edit machine configurations to do the following changes:
+
+- In both `controlplane.yaml` and `worker.yaml`, edit `.spec.machine.install.disk` to be `/dev/vda`
+  ```yaml
+  spec:
+    machine:
+      install:
+        disk: /dev/vda
+  ```
```

Comment:
I also think that configuring kubernetes API endpoint as VIP endpoint is not a good idea.
Don't remember the details but it can be chicken and egg problem.

Reply:
Ideally it should be loadbalancer endpoint.

Reply:
Hm, I thought it was one of the common practices of doing it as documented here: https://www.talos.dev/v1.1/talos-guides/network/vip/

I think what you meant is the caveat noted on that page.

Reply:
Ok, was a misunderstanding, addressed in https://github.com/siderolabs/talos/pull/5907#discussion_r922170222